### PR TITLE
Kubernetes manifests

### PR DESCRIPTION
**Pull request by narun4sk**
**Description of the change**

Divided `test.yaml` configuration into separate Kubernetes manifests.

**Benefits**

Now it's easier for the end users to adjust the configuration to their specific needs.

For example the persistence settings can be edited in the *volumes* manifest, essential environment variables in the *configs* & *secrets*, etc.

**Possible drawbacks**

None

**Additional information**

You might be tempted to replace the mainstream `redis` and `postgres` container images in the manifests with the *bitnami* equivalents, but I would like to recommend against it, as it creates a fake impression of the vendor lock, that the `bitnami-docker-discourse` would only work alongside other *bitnami* docker images.

### CODE REVIEW

**Comment by narun4sk** [comment]
In a nutshell:
- restored test.yaml
- crammed all k8s files into one manifest
- added comments (not sure if correct ones?)
- added more guidelines re Kubernetes

**Comment by javsalgar** [comment]
Thanks, @juan131 could you please review again?

File: README.md [hunk 1]
```diff
@@ -27,16 +27,39 @@ $ docker-compose up -d
...
 
+> NOTE: This will create `discourse-bitnami` namespace and deploy all the components in there.
+
+> NOTE: In the example configuration we are using `hostPath` driver, which is only suitable in the *sigle-node* cluster. Please change the driver to one which suits your environment better. See [Types of Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes) for more info.
+
 > NOTE: If you are pulling from a private containers registry, replace the image name with the full URL to the docker image. E.g.
 >
 > - image: 'your-registry/image-name:your-version'
 
+If you wanna monitor *Discourse* installation progress please run the following:
```
Comment (juan131) [inline README.md]: Avoid using informal expressions such as "wanna".

File: README.md [hunk 2]
```diff
@@ -27,16 +27,39 @@ $ docker-compose up -d
...
+
+```bash
+$ DISCOURSE_POD=$(kubectl -n discourse-bitnami get po | awk '/discourse/{print $1}')
+$ kubectl logs $DISCOURSE_POD discourse -f
+```
```
Comment (juan131) [inline README.md]: This very complex... I'd use sth like:

```bash
$ kubectl logs -f $(kubectl -n discourse-bitnami get pods -o jsonpath='{.items[0].metadata.name}')
```

File: README.md [hunk 3]
```diff
@@ -27,16 +27,39 @@ $ docker-compose up -d
...
+
+If you wanna test successfull *Discourse* installation, please try the following and navigate your browser to http://127.0.0.1:3000.
```
Comment (juan131) [inline README.md]: You can access your Discourse installation using [Port Forwarding](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster) as show below:

File: k8s/manifests.yaml [hunk 4]
```diff
@@ -0,0 +1,561 @@
+#
+# Create a Namespace, where Discourse & friends will be deployed.
```
Comment (juan131) [inline k8s/manifests.yaml]: `Discourse & friends` -> `Discourse related K8s resources`

File: k8s/manifests.yaml [hunk 5]
```diff
@@ -0,0 +1,561 @@
...
+#
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: discourse-bitnami
+---
+#
+# Create PersistentVolume, where Discourse, PostgreSQL
+# and Redis gonna keep the data.
```
Comment (juan131) [inline k8s/manifests.yaml]: Avoid using informal expressions such as "gonna".

**Comment by tompizmor** [comment]
@narun4sk Did you have time to check the latest Juan's suggestions?

**Comment by narun4sk** [comment]
@tompizmor Yes I did, probably will push the changes later today. Anyway, all the required changes are to do with the _informal language_, so I was expecting that someone from the bitnami crew can handle this.

**Comment by narun4sk** [comment]
@tompizmor @juan131 please take a look at 9e5da00

**Comment by tompizmor** [comment]
@narun4sk Shouldn't you be deleting the `test.yaml` file? My understanding is that `./k8s/manifests.yaml` is the replacement no?

Also, I see there are a lot of blank lines in the manifests.yaml. Is that on purpose? I would delete all of them.

**Comment by narun4sk** [comment]
@tompizmor This too was my idea, but please see the comment: https://github.com/bitnami/bitnami-docker-discourse/pull/69#pullrequestreview-204680457

Regarding blank lines, `yaml` has been made for humans, therefore readability is important. IMHO it's easier to identify different sections in the manifest when it's not crammed, therefore blank lines were added intentionally.

**Comment by tompizmor** [comment]
Ahh, that's right! Then I would have added your changes to the _test.yaml_ file instead of creating the new file (_k8s/manifests.yaml_). I am afraid that having the two files in the repository would be a bit confusing.
Do you agree @juan131?

**Comment by juan131** [comment]
I guess it can be confusing @tompizmor, we can keep all the changes on a single file if you'd rather do it that way.

**Comment by juan131** [comment]
ping @tompizmor

**Comment by tompizmor** [comment]
I would create one single file yes

**Pull request closed without merging**